File: iOS_app/Home Automation App/Models/Device.swift
//
//  Device.swift
//  Home Automation App
//
//  Created by Xavier Bou on 4/7/20.
//  Copyright © 2020 Xavier Bou. All rights reserved.
//

import Foundation

struct Device: Codable {
    var _id: String?
    var name: String?
    var deviceType: String?
    var state: String?
    var room: String?
    var receivingTopic: String?
    var respondingTopic: String?
    init() {
        _id = nil
        name = nil
        deviceType = nil
        state = nil
        room = nil
        receivingTopic = nil
        respondingTopic = nil
    }
}

struct DevicesList: Codable {
    var device: [Device]?
}

struct SingleDevice: Codable {
    let device: Device?
    init() {
        device = nil
    }
}

File: iot_HTTP_server/models/device.js
'use strict'

const mongoose = require('mongoose')
const Schema = mongoose.Schema

const DeviceSchema = Schema({
  name: String,
  deviceType: String,
  state: String,
  room: String,
  receivingTopic: String,
  respondingTopic: String
})

module.exports = mongoose.model('Device', DeviceSchema)
